### DISCUSSION TITLE
Reply error code along with spqr router internal error message

### DISCUSSION THREAD

**Discussion by reshke** [pg-sharding/spqr#375] [Ideas]
we can define method just alike https://github.com/pg-sharding/spqr/blob/master/pkg/client/client.go#L31 to reply PostgreSQL proto error by code, not by message.

```
func (cl *PsqlClient) ReplyErrMsgByCode(errcode string) error {
        clerrmsg := getErrMsgByCode(errcode)
	for _, msg := range []pgproto3.BackendMessage{
		&pgproto3.ErrorResponse{
			Message:  clerrmsg,
			Severity: "ERROR",
                        Code: errcode
		},
		&pgproto3.ReadyForQuery{
			TxStatus: byte(txstatus.TXIDLE),
		},
	} {
		if err := cl.Send(msg); err != nil {
			return err
		}
	}

	return nil
}
```

https://github.com/postgres/postgres/blob/master/src/backend/utils/errcodes.txt#L482-L486
we can use 

SPQRD for `failed to match any datashard`
SPQRE for `skip executing this query, wait for next`
SPQRC for ComplexQuery
SPQRF for FailedToMatch
SPQRS for SkipColumn
SPQRM for ShardingKeysMissing
SPQRX for CrossShardQueryUnsupported

**Comment by CBists** [comment]
SPQRR for `Routing error`
SPQRU for `Unexpected error`

**Comment by Denchick** [comment]
The feature was merged in #408
